src/cat/cat.rs

```diff
@@ -300,6 +319,52 @@ fn write_fast(files: Vec<String>) -> CatResult<()> {
         _ => Err(CatError::EncounteredErrors(error_count)),
     }
 }
+#[cfg(any(target_os = "linux", target_os = "android"))]
+fn write_fast(files: Vec<String>) -> CatResult<()> {
+    const BUF_SIZE: usize = 1024 * 16;
+    let writer = stdout();
+    // let _handle = writer.lock();
+    let (pipe_rd, pipe_wr) = pipe().unwrap();
+    let mut error_count = 0;
+
+    for file in files {
+        match open(&file[..]) {
+            Ok(handle) => loop {
+                let res = splice(
+                    handle.file_descriptor,
+                    None,
+                    pipe_wr,
+                    None,
+                    BUF_SIZE,
+                    SpliceFFlags::empty(),
+                ).unwrap();
```

Comment:
I'd prefer if you avoid the `unwrap()` here (and below).

Reply:
How would you suggest doing it differently? Perhaps with the following?:
```
.unwrap_or_else(|error| {
    eprintln!("{}", error);
    error_count += 1;
});
```